RInAppMessaging/Classes/Models/UserInfoProvider.swift

```diff
@@ -0,0 +1,75 @@
+/**
+ * Interface which client app should implement in order for InAppMessaging SDK to get information
+ * when needed.
+ */
+@objc public protocol UserInfoProvider {
+
+    /**
+     * Only return ID token if user is logged in. Else return null.
+     *
+     * @return String of ID token.
+     */
+    func getIDToken() -> String?
+
+    /**
+     * Only return user ID used when logging if user is logged in in the current session.
+     *
+     * @return String of the user ID.
+     */
+    func getUserId() -> String?
+
+    /**
+     * Only return Rakuten ID used in the current session.
+     *
+     * @return String of the Rakuten ID.
+     */
+    func getRakutenId() -> String?
+}
+
+// MARK: - Private Properties
+extension UserInfoProvider {
+    /// Converted preferences object from
+    /// an array that can be sent to the backend.
+    /// - Returns: A list of IDs to send in request bodies.
+    var userIdentifiers: [UserIdentifier] {
+        var userIdentifiers = [UserIdentifier]()
+
+        if let rakutenId = getRakutenId() {
+            userIdentifiers.append(
+                UserIdentifier(type: .rakutenId, identifier: rakutenId)
+            )
+        }
+
+        if let userId = getUserId() {
+            userIdentifiers.append(
+                UserIdentifier(type: .userId, identifier: userId)
+            )
+        }
+
```

Comment:
Should this be calling getIDToken() too?

Reply:
This part was transplanted from [`IAMPreferenceRepository`](https://github.com/rakutentech/ios-inappmessaging/blob/master/RInAppMessaging/Classes/Repositories/IAMPreferenceRepository.swift#L21-L29) (also see Android analog [`RuntimeUtil.kt:getUserIdentifiers()`](https://github.com/rakutentech/android-inappmessaging/blob/master/inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/RuntimeUtil.kt#L79-L84)) so I think the logic is correct